juju-tutorials/how-to-fixing-attach-volume-to-a-k8s-instance.md

```diff
@@ -0,0 +1,336 @@
+
+```bash
+juju config openstack-integrator auth-url=$OS_AUTH_URL region=$OS_REGION_NAME username=$OS_USERNAME user-domain-name=$OS_USER_DOMAIN_NAME password=$OS_PASSWORD project-name=$OS_PROJECT_NAME  project-domain-name=$OS_PROJECT_DOMAIN_NAME
+```
+
+### kubectl
+
+
+##### install
+
+The most common way to install `kubectl` in ubuntu is `snap`.
+
+```bash
+sudo snap install kubectl --classic
+```
+
+##### set up
+
+To configure kubectl, you must copy the configuration file from the deployed charmed kubernetes.
+
+```bash
+mkdir ~/.kube
+juju scp kubernetes-master/0:/home/ubuntu/config ~/.kube/config
+```
+[note status="VPN"]
+If your Openstack is behind a VPN, you may need to run `sshuttle` to create a tunnel for copying from the juju unit. `sshuttle -r <user@server> <juju-network>` e.g. `sshuttle -r bastion 10.5.0.0/24` (find network with `juju status -m controller`)
+[/note]
+
+### cinder volume
+
+##### create
+
+The following yaml configuration file is used to create an 8 GB cinder volume.
+
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: pvc-storage
+spec:
+  storageClassName: cdk-cinder
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 8Gi
+```
+
+The next step will be to create PersistentVolumeClaim (PVC).
+
+```bash
+kubectl apply -f <pvc-configutation-file>.yaml
+```
+
+##### attach
+
+This step shows how to attach the PVC created in the previous step. For simplicity, we will use nginx.
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: test-nginx
+spec:
+  restartPolicy: Never
+  volumes:
+    - name: test-volume
+      persistentVolumeClaim:
+        claimName: pvc-storage
+
+  containers:
+    - name: nginx-container
+      image: nginx
+      volumeMounts:
+      - name: test-volume
+        mountPath: /usr/share/nginx/html/test_volume
+```
+
+Run the command below to deploy a nginx container. 
+
+```bash
+kubectl create -f <nginx_config>.yaml
+```
+[note status="Proxy"]
+A proxy must be set on OpenStack.
+```bash
+juju config containerd http_proxy=http://squid.internal:3128 https_proxy=http://squid.internal:3128
+```
+[/note]
+
+### update QEMU
+
+TODO
+
+## Nova Compute - Fixing Attach Volume to Instance
+
+Duration: 20:00
+
+### Overview
+
+When updating **QEMU** packages, instances that started prior to the upgrade can experience issues attaching volumes to it. 
+This is due to [LP#1847361][#1847361]. In summary, this bug causes the new **QEMU** version to not recognize the old **QEMU** modules for attaching volumes to VMs. 
+This bug is especially bad in Kubernetes deployments on top of Openstack. A pod will get stuck attempting to create container when it tries to attach the PVC (and by consequence, attach cinder volume to k8s worker VM). This playbook intends to explain how to identify if the **QEMU** bug is the root problem, how to fix, and proper procedure for handling the k8s situation.
+
+### How To Identify
```

Comment:
I know this section name comes from the playbook. A better section title could be "Identify attachment failures"